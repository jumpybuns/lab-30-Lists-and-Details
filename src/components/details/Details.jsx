import React from 'react';
import PropTypes from 'prop-types';

function Details({ name, image, characters }) {
  const details = characters.map((character) => {
    <li key={character.name}>{details}</li>;
  });

  return (
    <>
      <figure>
        <img src={image} alt={name} />
        <figcaption>
          <p>{name}</p>
        </figcaption>
      </figure>
    </>
  );
}

export default Details;

Details.propTypes = {
  characters: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
