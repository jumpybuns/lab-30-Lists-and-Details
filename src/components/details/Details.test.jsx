import { BrowserRouter as Router } from 'react-router-dom';

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Details from './Details.jsx';

describe('Details component', () => {
  afterEach(() => cleanup());
  it('renders Details', () => {
    const { asFragment } = render(
      <Router>
        <Details />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
