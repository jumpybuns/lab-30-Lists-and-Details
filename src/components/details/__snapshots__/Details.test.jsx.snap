// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `
<DocumentFragment>
  <figure>
    <img />
    <figcaption>
      <p />
    </figcaption>
  </figure>
</DocumentFragment>
`;
